### PR TITLE
Add support for enabling autoscaling

### PR DESCRIPTION
**Pull request by vboulineau**
#### What this PR does / why we need it:

Add support for enabling autoscaling

#### Special notes for your reviewer:

Variable is not yet documented on purpose. Once `datadog-crds` chart is merged, we need to upgrade again the deps in `datadog-operator` and `datadog-agent`

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [x] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

**Review by clamoriniere — COMMENTED** [review]
Looks good.
could you add a scenario in https://github.com/DataDog/helm-charts/tree/main/charts/datadog/ci to validate that the option is working?

**Comment by vboulineau** [comment]
> Looks good. could you add a scenario in https://github.com/DataDog/helm-charts/tree/main/charts/datadog/ci to validate that the option is working?

Added

File: charts/datadog/requirements.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ dependencies:
   - name: datadog-crds
     version: 1.0.1
     repository: https://helm.datadoghq.com
-    condition: clusterAgent.metricsProvider.useDatadogMetrics
+    condition: datadog.autoscaling.workload.enabled,clusterAgent.metricsProvider.useDatadogMetrics
```
Comment (clamoriniere) [inline charts/datadog/requirements.yaml:5]: dummy question: `condition` is using "OR" operator between the fields?

File: charts/datadog/templates/cluster-agent-deployment.yaml [hunk 2]
```diff
@@ -344,6 +344,10 @@ spec:
             value: {{ .Values.datadog.prometheusScrape.version | quote }}
           {{- end }}
           {{- end }}
+          {{- if (((.Values.datadog.autoscaling).workload).enabled) }}
```
Comment (clamoriniere) [inline charts/datadog/templates/cluster-agent-deployment.yaml:347]: should we also check remote-config enabled option?

File: charts/datadog/requirements.yaml (hunk 1 above)
Comment (vboulineau) [reply charts/datadog/requirements.yaml:5]: Unfortunately no, it's fallback mechanism: it takes the first field that **exists** (not that says `true`), hence it relies on the fact that `datadog.autoscaling.workload.enabled` does not exist in our default `values.yaml`.

File: charts/datadog/templates/cluster-agent-deployment.yaml (hunk 2 above)
Comment (vboulineau) [reply charts/datadog/templates/cluster-agent-deployment.yaml:347]: I have added a check in NOTES

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**